packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -0,0 +1,742 @@
+                                "children": []
+                            },
+                            {
+                                "sel": "g",
+                                "data": {
+                                    "ns": "http://www.w3.org/2000/svg",
+                                    "attrs": {
+                                        "id": "${id}4"
+                                    }
+                                },
+                                "children": []
+                            }
+                        ]
+                    },
+                    {
+                        "sel": "g",
+                        "data": {
+                            "ns": "http://www.w3.org/2000/svg",
+                            "attrs": {
+                                "id": "keith-diagram_sprotty_$${id}2",
+                                "transform": "translate(${size * 0.25}, ${size * 0.5})"
+                            },
+                            "class": {
+                                "selected": false
+                            }
+                        },
+                        "children": [
+                            {
+                                "sel": "g",
+                                "data": {
+                                    "ns": "http://www.w3.org/2000/svg",
+                                    "attrs": {
+                                        "id": "${id}5"
+                                    }
+                                },
+                                "children": [
+                                    {
+                                        "sel": "text",
+                                        "data": {
+                                            "ns": "http://www.w3.org/2000/svg",
+                                            "style": {
+                                                "dominant-baseline": "middle",
+                                                "font-family": "overpass, sans-serif",
+                                                "font-size": "${size * 0.5}",
+                                                "opacity": 1
+                                            },
+                                            "attrs": {
+                                                "x": 0,
+                                                "xml:space": "preserve",
+                                                "y": 0,
+                                                "lengthAdjust": "spacingAndGlyphs"
+                                            }
+                                        },
+                                        "text": "${numProxies}"
+                                    }
+                                ]
+                            }
+                        ],
+                        "key": "$${id}2"
+                    }
+                ],
+                "key": "$${id}"
+            }`);
```

Comment:
is json.parse the best solution here?

Reply:
As far as I can tell yes, I've outsourced the method as discussed